### DISCUSSION TITLE
`-t` option is tricky

### DISCUSSION THREAD

**Discussion by sbatial** [eza-community/eza#179] [Development]
So I don't think there is a way to make -t the same as in ls.
I don't say it is impossible.
But the best implementation I have right now does work but nevertheless eats the next argument, even if it starts with a `-`.

The core problem(s) seem(s) to be: `-t` is not really a shorthand in `ls` but a dedicated option which is equivalent to `--time=modified`. And our parser can currently not represent shortform-only options.

I'll have to think about what the best way forward is in my eyes but my gut is that the best way of getting such a feature to work is to make changes to the parser first to have it support short options without a long form.

**Comment by cafkafk** [comment]
> I'll have to think about what the best way forward is in my eyes but my gut is that the best way of getting such a feature to work is to make changes to the parser first to have it support short options without a long form.

This sounds non trivial, I changed the pr to a draft for now.

**Comment by sbatial** [comment]
As already mentioned over there: this should be much more doable when using clap #195

**Comment by sbatial** [comment]
Ok.
I'm looking into this right now (based on the clap branch) and I thought I got something wrong at first and we are diverging from `ls`.
But no - well, yes.. but not in the way I thought - from the `ls` manpage:
![image](https://github.com/eza-community/eza/assets/46973479/f92a3bd5-fa23-4043-86e2-605850ada39e)

So now the question becomes... Compatibility with ls or sensible options.

I'm currently tending towards the latter. ls is not the newest program anymore as already mentioned in #321.

And although I generally like backwards compatibility I feel like `eza` is drifting away from being a drop-in-replacement/was never intended to be one the first place.

So my thoughts are right now:
- have `-t` as a shorthand for `--time`
- have `modified` (another divergence from `ls` which uses `mtime`) as a default
- find a sensible shorthand for `--sort` (so something along the lines of `-s t` would be possible as an alternative)

**Comment by cafkafk** [comment]
> And although I generally like backwards compatibility I feel like `eza` is drifting away from being a drop-in-replacement/was never intended to be one the first place.

We discussed compatibility on element and one option that has been floating around is to have a flag that turns on a mode that tries to mimic ls, which is what lsd does.

**Reply by sbatial** [reply]
I had thought about the same thing so yea, I could imagine that :+1:
